### ISSUE TITLE
tslint 4.1.0 and try/catch results in 'expected variable-declaration: 'ex' to have a typedef (typedef)'

### ISSUE THREAD

**Issue by vladeck** [palantir/tslint#1880]
Using `typedef` rule as defined bellow:

```json
"typedef": [
    true,
    "call-signature",
    "arrow-call-signature",
    "parameter",
    "arrow-parameter",
    "property-declaration",
    "variable-declaration",
    "member-variable-declaration"
]
```

with this piece of code:

```typescript
try {
    // ...
} catch (ex) {
    // ...
}
```

Results in a warning from the title of this issue. Typescript does not allow defining type for `catch` clause parameter.

**Comment by adidahiya** [comment]
😢  this is a recurring issue we've had to fix multiple times. #145, #401.

Looks like a regression caused by https://github.com/palantir/tslint/commit/edeff3b3a8ee691b904e7b94866ec1789579425d#diff-4fc2495876ded0ebf40e7ccc7c9cbc66R188

Let's make sure to add regression tests for this in the next fix

**Comment by sanex3339** [comment]
please fix it as soon as possible, my travis builds are failing because this error
